src/lib/services/searchService.ts

```diff
@@ -86,8 +90,35 @@ export const useSearchHandler = (
   const { data: icnsAddressData, isFetching: icnsAddressFetching } =
     useAddressByICNSName(debouncedKeyword);
 
+  const [initiaAddr, moduleName, functionName] = splitModule(debouncedKeyword);
+
+  const enableModuleFetching = useMemo(
+    () =>
+      Boolean(
```

Comment:
the value is already a boolean no need to cast it again

Reply:
could be `boolean | "" | undefined` if you don't wrap it